compiler-core/src/parse/tests.rs

```diff
@@ -1002,6 +1002,34 @@ fn main() {
     );
 }
 
+#[test]
+fn case_list_pattern_after_spread() {
+    assert_module_error!(
+        "
+fn main() {
+    case [1, 2, 3] {
```

Comment:
let's remove the case expression as we're not testing anything to do with case expressions here.